File: trunk/GoServer/src/responsors/OnPacket_ReqChangeUsername.go
package responsors

import (
	"fmt"
	//"net"
	"github.com/golang/protobuf/proto"
	. "../frame"
	. "../packs"
)

func OnPacket_ReqChangeUsername(client *Client, pack proto.Message)() {
	//conn := client.Conn

	req := pack.(*ReqChangeUsername)
	fmt.Println("OnPacket_ReqChangeUsername")
	fmt.Println(req.Username)

	rsp := &ChangeUsername{req.Userid, req.Username}

	err := GetUserSystem().ChangeUsername(req.Userid, req.Username);
	CheckError(err)
	client.Send(rsp);
}

File: trunk/Server/HallAndRoomServer/OnPacket/OnPacket_ReqChangeUsername.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using MySql.Data.MySqlClient;

namespace HallAndRoomServer.OnPacket
{
    class OnPacket_ReqChangeUsername
    {
        public static void OnPacket(Client client, object pack)
        {
            ReqChangeUsername req = pack as ReqChangeUsername;
            ChangeUsername rsp = new ChangeUsername();
            rsp.Userid = req.Userid;
            rsp.Username = req.Username;
            client.Send(rsp);
            UserSystem.me.ChangeUsername(req.Userid, req.Username);
            return;
        }
    }
}
